packages/core/src/extensions/TextAlignment/TextAlignmentExtension.ts

```diff
@@ -0,0 +1,73 @@
+import { Extension } from "@tiptap/core";
+import { getBlockInfoFromPos } from "../Blocks/helpers/getBlockInfoFromPos";
+
+declare module "@tiptap/core" {
+  interface Commands<ReturnType> {
+    textAlignment: {
+      setTextAlignment: (
+        textAlignment: "left" | "center" | "right" | "justify"
+      ) => ReturnType;
+    };
+  }
+}
+
+export const TextAlignmentExtension = Extension.create({
+  name: "textAlignment",
+
+  addGlobalAttributes() {
+    return [
+      {
+        types: ["paragraph", "heading", "bulletListItem", "numberedListItem"],
```

Comment:
Is there a reason we set this on specific block types, instead of blockcontainer? If so, maybe good to comment this

Reply:
Yeah it's because applying it to `blockContainer` will apply it to all child blocks, which I think isn't great from a UX perspective. Will add comment